### DISCUSSION TITLE
not sure why I'm not receiving realtime updates for a table

### DISCUSSION THREAD

**Discussion by cocodrino** [supabase/supabase#1186] [Questions] [answered]
hi guys, right now I've this typescript code for a svelte store

```ts
import {writable} from "svelte/store";
import type {TodoType} from "../types/todo.type";
import supabase from "../SupabaseInstance";

const todoStore = writable<TodoType[]>([])

supabase.from('*')
    .on('*',(rsp)=>{
        console.info("RECEIVE ",rsp)
    })
    .subscribe()



supabase
    .from<TodoType>('todos')
    .on('*', payload => {
        console.log('Change received!', payload.new.title)
        todoStore.update((store)=>[...store,payload.new])
    })
    .subscribe()

const addTodo = async (title : string)=>{
    console.info("adding todo ", title)
    try {
        await supabase.from<TodoType>('todos')
            .insert({title: title, done: false})
    } catch (e) {
        console.error("ERROR ", e)
    }
}

const addTodos = (todos : TodoType[])=>{
    todoStore.set(todos)
}

export default {
    subscribe : todoStore.subscribe,
    addTodo,
    addTodos
}

```

addTodo is working fine, so It's clear that the supabase instance is working fine, also the table 'todos' exists and I can add new rows, also I can retrieve data 

```ts
let response = await supabase.from<TodoType>("todos").select();
        console.info("response is ",response)
        const todos = response.data;
        Todos.addTodos(todos)
```

but real-time updates are not working for me...do you know what could be the problem here?.

thank you so much

**Comment by kiwicopple** [comment, marked as answer]
Hey @cocodrino , is realtime turned on for your table? https://supabase.io/docs/guides/api#managing-realtime

**Reply by cocodrino** [reply]
🤦‍♂️ thank you....sorry for the noob question...yes, that fix the "problem" 🤦‍♂️

**Reply by kiwicopple** [reply]
cool!

**Reply by srvmux** [reply]
I was banging my head against my desk for an hour trying to figure out why it wasn't working, it was returning an empty response, probably should have some sort of an indication of when you try to use realtime events when they're not enable on the backend, like throw and error or something like that.

thanks for the great work

**Reply by clouedoc** [reply]
I had the same problem too. The backend should throw an error when realtime is disabled.
